api.graphql

```diff
@@ -0,0 +1,75 @@
+# Proposed interface
+# API gets 'msg' and 'medium'
+# 
+
+schema {
+  query: Query
+}
+
+# Message, Mode: (TARGET, ANNOUNCE)
+# - Announce:
+#   - Facebook, Twitter, Slack @announcements, Live Site
+# - Target:
+#   - Email, Twilio, Slacking PMs
+# message, group, and medium
+# for each medium, we have a mapping to a group
+# Email: - select user list via tags
+# Mobile: - same thing as email
+# Slack: - send to general with prepended/preformatted [TAG NAME]
+#        - or direct PMs
+# Tweet: No targetting - ( provide disclaimer )
+# Push notification: - can we target? api tokens
+type Query {
+  send_message(message: String!, plugins: PluginMaster!): PluginReturn!
+}
+
+#
+# enum StatusEnum {
+#   SUCCESS
+#   ERROR
+#   NULL
+# }
+
+type Status { # corresponding to APIReturn
+  error: Int!,
+  debugInfo: String
+}
+
+input PluginMaster {
+  email: EmailConfig,
+  facebook: FacebookConfig,
+  live_site: LiveSiteConfig,
+  twitter: TwitterConfig,
+  slack: SlackConfig
+}
+
+type PluginReturn {
+  email: Status,
+  facebook: Status,
+  live_site: Status,
+  twitter: Status,
+  slack: Status
+}
+
+# Dummies
+input EmailConfig {
+  name: String!
```

Comment:
Does GraphQL not allow empty types? e.g. `input EmailConfig {}`, if it does we should remove members that we are not using. Since GraphQL isn't versioned we have to be really careful to change the schema once we make one.